apisix/plugins/server-info.lua

```diff
@@ -158,25 +159,65 @@ local function report(premature, report_ttl)
         end
     end
 
-    server_info.last_report_time = ngx_time()
-
+    -- get inside etcd data, if not exist, create it
     local key = "/data_plane/server_info/" .. server_info.id
-    local ok, err = core.etcd.set(key, server_info, report_ttl)
+    local res, err = core.etcd.get(key)
+    if err ~= nil then
+        core.log.error("failed to get server_info from etcd: ", err)
+    end
+
+    if  not res.body.node then
+        local res_new, err = core.etcd.set(key, server_info, report_ttl)
```

Comment:
I think the logic for initial setting can be merged to the below logic (`modified_version` is zero).

Reply:
So we don't have two similar code blocks.

Reply:
Can we set `modified_version ` to `nil`, I'm not sure if it can be set to nil.

Reply:
I have already said you can set it to `0`.

Reply:
updated it.